packages/core/src/baseclient.ts

```diff
@@ -618,7 +618,7 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
           throw new SentryError('`beforeSend` returned `null`, will not send event.');
         }
 
-        const session = scope && scope.getSession && scope.getSession();
+        const session = scope && scope.getSession();
```

Comment:
This doesn't have anything to do with baggage but it's a no longer necessary check I stumbled across. See [this conversation](https://github.com/getsentry/sentry-javascript/pull/5054#discussion_r869359697) for context

Reply:
Btw, why not use `scope?.getSession()` notation?

Reply:
Great question!

It's because we try to optimize our codebase around targeting `es6`. Optional Chaining is a newer JS feature, `ES2020 - "ES11"` as per: https://gist.github.com/rajaramtt/7df3702a04c644b0b62c9a64f48f3dbf#64-optional-chaining. This means that it'll have to be down-compiled to `es6` compatible code when we generate the final assets to distribute on npm and the CDN (this is for max browser + node version support).

Something like so: 

```js
if (hey?.me) {
 console.log('me'); 
}
```

ends up getting down-compiled to:

```js
if (hey !== null && hey !== void 0 && hey.me) {
  console.log('me');
}
```

[See the babel demo](https://babeljs.io/repl#?browsers=defaults&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=MYewdgzgLgBAFgUwJ4wLwwN4F8DcAoPASwDMYAKRFAMivmQDoBbBASkzxlEhABsF6eIAOZkA5M1EscMPFgIlylAPxNW7TuAi9-gkeISTpsoA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.18.4&externalPlugins=&assumptions=%7B%7D).

which is way more bytes, which is bad for bundle size.

We made a decision as a result to not use any optional chaining in code that might be used in browser environments. See some examples on bundle wins we made here: https://github.com/getsentry/sentry-javascript/pulls?q=is%3Apr+optional+chaining+is%3Aclosed+milestone%3A%22Tree+shaking+%2F+Bundle+Size%22